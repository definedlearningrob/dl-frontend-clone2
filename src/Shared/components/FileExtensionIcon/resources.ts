export const imageExtensions = ['jpg', 'jpeg', 'png', 'gif'];
export const videoExtensions = ['wma', 'mpg', 'flv', 'avi', 'mp4', 'webm'];
export const documentExtensions = ['pdf', 'doc', 'docx', 'txt', 'csv', 'rtf', 'html'];
export const presentationExtensions = ['ppt', 'pptx'];
export const sheetExtensions = ['xls', 'xlsx'];
