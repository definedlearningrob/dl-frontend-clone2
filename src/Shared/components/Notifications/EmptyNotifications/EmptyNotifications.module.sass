@import '@shared/stylesheets/config.sass'

.emptyState
  +flex-container(column, center, center)
  height: 100%
  overflow: hidden
  width: 100%

.emptyStateImage
  width: 60%

.emptyStateText
  color: $color-primary-500
  font-family: $font-family-body
  font-size: $font-size-base
  font-stretch: normal
  font-style: normal
  font-weight: $font-weight-medium
  line-height: $line-height-base
