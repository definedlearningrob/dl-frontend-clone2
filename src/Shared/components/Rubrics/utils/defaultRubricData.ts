export const defaultRubricData = {
  headings: [
    { id: 'heading-1', name: 'rubric.careerPathway.name', multiplier: 1 },
    { id: 'heading-2', name: 'rubric.projectGoal.name', multiplier: 1 },
    { id: 'heading-3', name: 'rubric.projectRole.name', multiplier: 1 },
    { id: 'heading-4', name: 'rubric.jobProductQuality.name', multiplier: 1 },
  ],
  criteriaLabels: [
    { displayName: 'test label', id: 'criteria-label-1', score: 1 },
    { displayName: null, id: 'criteria-label-2', score: 2 },
    { displayName: null, id: 'criteria-label-3', score: 3 },
    { displayName: null, id: 'criteria-label-4', score: 4 },
  ],
  criterias: [
    {
      id: 'criteria-1-1',
      rubricCriteriaLabelId: 'criteria-label-1',
      rubricHeadingId: 'heading-1',
      text: 'rubric.careerPathway.1',
    },
    {
      id: 'criteria-1-2',
      rubricCriteriaLabelId: 'criteria-label-2',
      rubricHeadingId: 'heading-1',
      text: 'rubric.careerPathway.2',
    },
    {
      id: 'criteria-1-3',
      rubricCriteriaLabelId: 'criteria-label-3',
      rubricHeadingId: 'heading-1',
      text: 'rubric.careerPathway.3',
    },
    {
      id: 'criteria-1-4',
      rubricCriteriaLabelId: 'criteria-label-4',
      rubricHeadingId: 'heading-1',
      text: 'rubric.careerPathway.4',
    },
    {
      id: 'criteria-2-1',
      rubricCriteriaLabelId: 'criteria-label-1',
      rubricHeadingId: 'heading-2',
      text: 'rubric.projectGoal.1',
    },
    {
      id: 'criteria-2-2',
      rubricCriteriaLabelId: 'criteria-label-2',
      rubricHeadingId: 'heading-2',
      text: 'rubric.projectGoal.2',
    },
    {
      id: 'criteria-2-3',
      rubricCriteriaLabelId: 'criteria-label-3',
      rubricHeadingId: 'heading-2',
      text: 'rubric.projectGoal.3',
    },
    {
      id: 'criteria-2-4',
      rubricCriteriaLabelId: 'criteria-label-4',
      rubricHeadingId: 'heading-2',
      text: 'rubric.projectGoal.4',
    },
    {
      id: 'criteria-3-1',
      rubricCriteriaLabelId: 'criteria-label-1',
      rubricHeadingId: 'heading-3',
      text: 'rubric.projectRole.1',
    },
    {
      id: 'criteria-3-2',
      rubricCriteriaLabelId: 'criteria-label-2',
      rubricHeadingId: 'heading-3',
      text: 'rubric.projectRole.2',
    },
    {
      id: 'criteria-3-3',
      rubricCriteriaLabelId: 'criteria-label-3',
      rubricHeadingId: 'heading-3',
      text: 'rubric.projectRole.3',
    },
    {
      id: 'criteria-3-4',
      rubricCriteriaLabelId: 'criteria-label-4',
      rubricHeadingId: 'heading-3',
      text: 'rubric.projectRole.4',
    },
    {
      id: 'criteria-4-1',
      rubricCriteriaLabelId: 'criteria-label-1',
      rubricHeadingId: 'heading-4',
      text: 'rubric.jobProductQuality.1',
    },
    {
      id: 'criteria-4-2',
      rubricCriteriaLabelId: 'criteria-label-2',
      rubricHeadingId: 'heading-4',
      text: 'rubric.jobProductQuality.2',
    },
    {
      id: 'criteria-4-3',
      rubricCriteriaLabelId: 'criteria-label-3',
      rubricHeadingId: 'heading-4',
      text: 'rubric.jobProductQuality.3',
    },
    {
      id: 'criteria-4-4',
      rubricCriteriaLabelId: 'criteria-label-4',
      rubricHeadingId: 'heading-4',
      text: 'rubric.jobProductQuality.4',
    },
  ],
};
