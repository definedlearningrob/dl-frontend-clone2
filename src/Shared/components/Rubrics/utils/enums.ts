export enum RUBRIC_TYPE {
  VIEWER = 'VIEWER',
  GRADER = 'GRADER',
  PREVIEW = 'PREVIEW',
}
