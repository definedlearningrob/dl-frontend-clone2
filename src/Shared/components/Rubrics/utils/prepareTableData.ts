import { v4 as uuid } from 'uuid';

import {
  FillMissingCriteriasArguments,
  RubricHeadingIdPairs,
  RubricCriteria,
  RubricData,
} from './types';

const fillMissingCriteria = ({
  headings,
  criteriaLabels,
  criterias,
}: FillMissingCriteriasArguments): RubricCriteria[] => {
  const labelIds = criteriaLabels.map((label) => label.id);
  const headingIds = headings.map((heading) => heading.id);

  const labelHeadingIdPairs: RubricHeadingIdPairs = labelIds.reduce(
    (acc: RubricHeadingIdPairs, criteriaLabelId) => {
      const pairsForFirst = headingIds.map((headingId) => ({ headingId, criteriaLabelId }));
      acc.push(...pairsForFirst);

      return acc;
    },
    []
  );

  return labelHeadingIdPairs.map((pair) => {
    const foundCriteria = criterias.find(
      ({ rubricCriteriaLabelId, rubricHeadingId }) =>
        rubricCriteriaLabelId === pair.criteriaLabelId && rubricHeadingId === pair.headingId
    );

    return (
      foundCriteria || {
        uuid: uuid(),
        text: '',
        rubricCriteriaLabelId: pair.criteriaLabelId,
        rubricHeadingId: pair.headingId,
      }
    );
  });
};

export const prepareTableData = (rubric: RubricData): RubricData => {
  const { criterias, headings, criteriaLabels } = rubric;

  return {
    ...rubric,
    criterias: fillMissingCriteria({ headings, criteriaLabels, criterias }),
  };
};
