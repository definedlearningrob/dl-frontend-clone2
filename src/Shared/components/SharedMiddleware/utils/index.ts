export * from './parseSharedResource';
export * from './getResourceLink';
