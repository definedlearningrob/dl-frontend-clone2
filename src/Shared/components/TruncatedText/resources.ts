export const OFFSET_HEIGHT = 63;
