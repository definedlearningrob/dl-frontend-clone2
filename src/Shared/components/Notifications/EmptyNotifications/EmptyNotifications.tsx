import { useTranslation } from 'react-i18next';

import notificationEmptyImage from '@shared/assets/images/notification-empty.png';
import { NOTIFICATION_TYPES } from '@shared/resources/constants';

import styles from './EmptyNotifications.module.sass';

type Props = {
  type?: NOTIFICATION_TYPES;
};

export const EmptyNotifications = ({ type }: Props) => {
  const { t } = useTranslation();
  const translationKey =
    type === NOTIFICATION_TYPES.ANNOUNCEMENT
      ? 'appHeader.announcements'
      : 'appHeader.notifications';

  return (
    <div className={styles.emptyState}>
      <img
        alt={styles.emptyStateText}
        className={styles.emptyStateImage}
        src={notificationEmptyImage}
      />
      <p className={styles.emptyStateText}>{t(`${translationKey}.emptyState`)}</p>
    </div>
  );
};
