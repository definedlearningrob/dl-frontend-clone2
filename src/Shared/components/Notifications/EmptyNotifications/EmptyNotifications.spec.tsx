import { screen } from '@testing-library/react';

import { renderWithI18N } from '@pbl/utils/test';

import { EmptyNotifications } from '@shared/components/Notifications/EmptyNotifications/EmptyNotifications';
import { NOTIFICATION_TYPES } from '@shared/resources/constants';

describe('EmptyNotifications', () => {
  it('should render with no type passed', () => {
    const { container } = renderWithI18N(<EmptyNotifications />);

    screen.getByText('No notifications yet!');
    expect(container).toMatchSnapshot();
  });
  it('should render with announcement type passed', () => {
    const { container } = renderWithI18N(
      <EmptyNotifications type={NOTIFICATION_TYPES.ANNOUNCEMENT} />
    );

    screen.getByText('No announcements yet!');
    expect(container).toMatchSnapshot();
  });
});
