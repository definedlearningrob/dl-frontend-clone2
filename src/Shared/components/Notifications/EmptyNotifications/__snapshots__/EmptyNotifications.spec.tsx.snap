// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyNotifications should render with announcement type passed 1`] = `
<div>
  <div
    class="emptyState"
  >
    <img
      alt="emptyStateText"
      class="emptyStateImage"
      src="notification-empty.png"
    />
    <p
      class="emptyStateText"
    >
      No announcements yet!
    </p>
  </div>
</div>
`;

exports[`EmptyNotifications should render with no type passed 1`] = `
<div>
  <div
    class="emptyState"
  >
    <img
      alt="emptyStateText"
      class="emptyStateImage"
      src="notification-empty.png"
    />
    <p
      class="emptyStateText"
    >
      No notifications yet!
    </p>
  </div>
</div>
`;
